packages/2d/src/components/Icon.ts

```diff
@@ -0,0 +1,89 @@
+import {Image, ImageProps} from '../components/Image';
+import {colorSignal, computed, initial, signal} from '../decorators';
+import {SignalValue, SimpleSignal} from '@motion-canvas/core/lib/signals';
+import {useLogger} from '@motion-canvas/core/lib/utils';
+import {ColorSignal} from '@motion-canvas/core/lib/types';
+
+export interface IconProps extends ImageProps {
+  /**
+   * Expects an Icon Identifier.
+   * You can find identifiers on [Icônes](https://icones.js.org/collection/all).
+   * They can look like this:
+   * * mdi:language-typescript
+   * * ph:anchor-simple-bold
+   * * ph:activity-bold
+   */
+  icon: SignalValue<string>;
+
+  /**
+   * Color of the Icon. Defaults to white.
+   * Provide the color in one of the following formats:
+   * * named color like `red`, `darkgray`, …
+   * * hexadecimal string with # `#bada55`,`#141414`
+   * Values can be either RGB or RGBA: `#bada55` or `#bada55aa` (latter is partially transparent)
+   * The shorthand version (e.g. `#abc` for `#aabbcc`) is also possible
+   */
+  color?: SignalValue<string>;
+}
+
+/**
+ * An Icon Component that provides an easy access to over 150k icons.
+ * See https://icones.js.org/collection/all for all available Icons.
+ */
+export class Icon extends Image {
+  /**
+   * Expects an Icon Identifier.
+   * You can find identifiers on [Icônes](https://icones.js.org).
+   * They can look like this:
+   * * mdi:language-typescript
+   * * ph:anchor-simple-bold
+   * * ph:activity-bold
+   */
+  @signal()
+  public declare icon: SimpleSignal<string, this>;
+
+  /**
+   * Color of the Icon. Defaults to white.
+   * Provide the color in one of the following formats:
+   * * named color like `red`, `darkgray`, …
+   * * hexadecimal string with # `#bada55`,`#141414`
+   * Values can be either RGB or RGBA: `#bada55` or `#bada55aa` (latter is partially transparent)
+   * The shorthand version (e.g. `#abc` for `#aabbcc`) is also possible
+   */
+  @initial('white')
+  @colorSignal()
+  public declare color: ColorSignal<this>;
```

Comment:
Even though there's no corresponding `stroke` property (which would be an awesome feature for icons but that's a feature for another day), I'd say that this should be called `fill` to be consistent with the majority of the other components.

Reply:
`Image`, the parent class of `Icon`, uses `fill` to set the color of the background image, so I don't think that is a good idea.

See for example:
```jsx
<Layout layout direction={"row"}>
      <Icon icon="mdi:alert-minus" color={"red"} size={200} fill={"yellow"} />
      <Image src={"https://upload.wikimedia.org/wikipedia/commons/2/2a/Alpha_compositing.svg"} size={[400, 200]} fill={"gray"} />
    </Layout>  
```
results in 

![image](https://user-images.githubusercontent.com/27686785/218324041-12a3aad6-f6ec-43e1-b7cd-b44c6ba1d941.png)

Reply:
ah ok, disregard then